src/docs/components/properties.md

```diff
@@ -333,6 +333,137 @@ export class ToDoListItem {
+})
+export class ToDoListItem {
+    // onComplete has an explicit type annotation
+    // of `any`, and no default value
+    @Prop() isComplete: any;
+    // label has an explicit type annotation of
+    // `any` with a default value of 'urgent',
+    // which is a string
+    @Prop() label: any = 'urgent';
+    // thingToDo has no type and no default value,
+    // and will be considered to be type `any` by
+    // TypeScript
+    @Prop() thingToDo;
+
+    render() {
+        return (
+            <ul>
+                <li>isComplete has a value of "{this.isComplete}" and a type of {typeof this.isComplete}</li>
+                <li>label has a value of "{this.label}" and a type of {typeof this.label}</li>
+                <li>thingToDo has a value of "{this.thingToDo}" and a type of {typeof this.thingToDo}</li>
+            </ul>
+        );
+    }
+}
+```
+
+When using a Stencil prop typed as `any` (implicitly or explicitly), the value that is provided to a prop retains its
+own type information. Neither Stencil nor TypeScript will try to change the type of the prop. To demonstrate, let's use
+`todo-list-item` twice, each with different prop values:
+
+```tsx
+{/* Using todo-list-item in TSX using differnt values each time */}
+<todo-list-item isComplete={42} label={null} thingToDo={"Learn about any-typed props"}></todo-list-item>
+<todo-list-item isComplete={"42"} label={1} thingToDo={"Learn about any-typed props"}></todo-list-item>
+```
+
+The following will rendered from the usage example above: 
+```md
+- isComplete has a value of "42" and a type of number 
+- label has a value of "" and a type of object 
+- thingToDo has a value of "Learn about any-typed props" and a type of string
+
+- isComplete has a value of "42" and a type of string
+- label has a value of "1" and a type of number 
+- thingToDo has a value of "Learn about any-typed props" and a type of string
+```
+
+In the first usage of `todo-list-item`, `isComplete` is provided a number value of 42, whereas in the second usage it 
+receives a string containing "42". The types on `isComplete` reflect the type of the value it was provided, 'number' and
+'string', respectively.
+
+Looking at `label`, it is worth noting that although the prop has a [default value](properties#default-values), it does
+not narrow the type of `label` to be of type 'string'. In the first usage of `todo-list-item`, `label` is provided a
+value of null, whereas in the second usage it receives a number value of 1. The types of the values stored in `label`
+are correctly reported as 'object' and 'number', respectively.
+
+#### Optional Types
+
+TypeScript allows members to be marked optional by appending a `?` at the end of the member's name. The example below
+demonstrates making each a component's props optional:
+
+```tsx
+import { Component, Prop, h } from '@stencil/core';
+
+@Component({
+    tag: 'todo-list-item',
+})
+export class ToDoListItem {
+    // onComplete is optional, has an explicit type
```

Comment:
```suggestion
    // isComplete is optional, has an explicit type
```

Reply:
Good catch  - changed this var name to `completeMsg` in https://github.com/ionic-team/stencil-site/pull/839/commits/311b99dbf1b270b8e80fe655029193a8573fa894